wazo_call_logd/plugins/cdr/schemas.py

```diff
@@ -93,6 +103,30 @@ class CDRSchema(Schema):
     tags = fields.List(fields.String(), attribute='marshmallow_tags')
     recordings = fields.Nested('RecordingSchema', many=True, default=[])
 
+    @post_dump(pass_original=True)
+    def _convert_destination_details_to_appropriate_schema(
+        self, data, original, **kwargs
+    ):
+        if original.destination_details:
+            destination_details_dict = dict()
+            for detail in original.destination_details:
+                key = detail.destination_details_key
+                value = detail.destination_details_value
+                if key == 'type':
+                    destination_details_dict['type'] = value
+                elif key == 'user_uuid':
+                    destination_details_dict['user_uuid'] = value
+                elif key == 'user_name':
+                    destination_details_dict['user_name'] = value
+                elif key == 'meeting_uuid':
+                    destination_details_dict['meeting_uuid'] = value
+                elif key == 'meeting_name':
+                    destination_details_dict['meeting_name'] = value
+                elif key == 'conference_id':
+                    destination_details_dict['conference_id'] = int(value)
```

Comment:
With one schema per destination, we avoid behavior of 
* `type: user`, `conference_id: 2`

The only way to know it's not possible, it's by reading the implementation of `CELGenUserEvent`.
We should make it explicit by design in the code too